glib-macros/src/lib.rs

```diff
@@ -265,6 +266,101 @@ pub fn clone(item: TokenStream) -> TokenStream {
     clone::clone_inner(item)
 }
 
+/// The same as [`gclosure_local!`](crate::gclosure_local) but uses [`Closure::new`] as a
+/// constructor.
+///
+/// [`Closure::new`]: https://gtk-rs.org/gtk-rs-core/stable/latest/docs/glib/closure/struct.Closure.html#method.new
+#[proc_macro]
+#[proc_macro_error]
+pub fn gclosure(item: TokenStream) -> TokenStream {
```

Comment:
Consensus in #343 is to remove the `g` prefix, so let's get this in as `closure!` and `closure_local!` directly.